Oneirocom/Magick
# Task optimizations

## What Changed:

Swicthed task plugin to use central store of tasks rather than hold them all in memory.  Moved our rete fork to internal so we can modify and have control of the source code.  Rete now lives in @magickml/rete.

## How to test:

Load a large spell, like Rob Lennons big ugly one, and watch your  system processes.  There should be no huge spikes in memory and things should remain stable.

Ensure agents work, and spells fire off nodes are light up the node graph.  This touches the task plugin, so should be well tested with a few different spells if possible.  Alternative, we push it to dev and test it there and patch as needed. This is a major fix needed to stabilize infrastructure.

## Additional information:

Any other information that might be useful while reviewing.
